Framework/Core/include/Framework/TMessageSerializer.h

```diff
@@ -13,41 +13,92 @@
 #include <fairmq/FairMQMessage.h>
 
 #include <TMessage.h>
-#include <TClonesArray.h>
+#include <TObjArray.h>
+#include <TList.h>
+#include <mutex>
+#include <TStreamerInfo.h>
+#include <memory>
 
 namespace o2 {
 namespace framework {
 
 class FairTMessage : public TMessage
 {
   public:
+    using TMessage::TMessage;
     FairTMessage(void* buf, Int_t len)
         : TMessage(buf, len)
     {
         ResetBit(kIsOwner);
     }
-};
 
-// helper function to clean up the object holding the data after it is transported.
-static void free_tmessage(void* /*data*/, void* hint)
-{
-    delete (TMessage*)hint;
-}
+    // helper function to clean up the object holding the data after it is transported.
+    static void free_fn(void* /*data*/, void* hint)
```

Comment:
since we are on it, nothing would prevent us from simply using `free`, it's a function anyhow, `_fn` is not more descriptive. Or, following the original naming, `freeFairTMessage`.

Reply:
changed to "free" since it already is member of FairTMessage